pulsar-broker/src/main/java/org/apache/pulsar/broker/web/PulsarWebResource.java

```diff
@@ -493,13 +502,47 @@ private URI getRedirectionUrl(ClusterData differentClusterData) throws Malformed
             } else {
                 serviceNameResolver.updateServiceUrl(differentClusterData.getServiceUrl());
             }
-            URL webUrl = new URL(serviceNameResolver.resolveHostUri().toString());
-            return UriBuilder.fromUri(uri.getRequestUri()).host(webUrl.getHost()).port(webUrl.getPort()).build();
+            return getConnectableURI(serviceNameResolver);
         } catch (PulsarClientException.InvalidServiceURL exception) {
             throw new MalformedURLException(exception.getMessage());
         }
     }
 
+    private CompletableFuture<URI> getConnectableURI(PulsarServiceNameResolver serviceNameResolver) {
+        return CompletableFuture.supplyAsync(() -> {
+            String hostUri = serviceNameResolver.resolveHostUri().toString();
+            String firstHostUri = hostUri;
+            boolean connectable = false;
+            URL webUrl = null;
+            do {
+                try {
+                    webUrl = new URL(hostUri);
+                    Socket socket = new Socket();
+                    socket.connect(new InetSocketAddress(webUrl.getHost(), webUrl.getPort()), 5000);
+                    socket.close();
+                    connectable = true;
+                } catch (Exception e) {
+                    log.warn("Unable connect to url:{}", hostUri);
+                    hostUri = serviceNameResolver.resolveHostUri().toString();
+
+                    if (firstHostUri.equals(hostUri)) {
+                        break;
+                    }
+                }
+            } while (!connectable);
+
+            if (webUrl == null) {
+                try {
+                    webUrl = new URL(hostUri);
+                } catch (MalformedURLException e) {
+                    throw new RestException(e);
+                }
+            }
+
+            return UriBuilder.fromUri(uri.getRequestUri()).host(webUrl.getHost()).port(webUrl.getPort()).build();
+        }, pulsar.getExecutor());
+    }
+
```

Comment:
We may add this logic to PulsarServiceNameResolver

Reply:
> We may add this logic to PulsarServiceNameResolver

This seems odd, since the method also needs to verify URI is an HTTP protocol